tools/nightly_checkout.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+"""Checks out the nightly development version of PyTorch and installs pre-built
+binaries into the repo.
```

Comment:
Can you  add usage instructions here @scopatz, and also in `CONTRIBUTING.md`?

Reply:
With a big warning probably: don't do this in your default/root conda env:)

Reply:
Or is that fine? I didn't see it create a new env, so hurriedly hit Ctrl-C

Reply:
Yeah, I did this too! Need to be more clear about what the tool is actually going to do, especially if it's going to do things like make new conda environments. A common situation is there's already an env a user wants to use, but not necessarily!

Reply:
It is totally fine to use this in the root conda-env.  I have added some prelim docs. Let me know if more are needed

Reply:
ie - this doesn't touch the base env in anyway